src/streamlink/plugins/tf1.py

```diff
@@ -17,21 +17,23 @@
 log = logging.getLogger(__name__)
 
 
-@pluginmatcher(re.compile(
-    r"https?://(?:www\.)?(?:tf1\.fr/([\w-]+)/direct|(lci)\.fr/direct)/?"
-))
+@pluginmatcher(re.compile(r"""
+    https?://(?:www\.)?(?:tf1\.fr/(?P<live>[\w-]+)/direct/?|
+    (?P<lci>tf1info|lci)\.fr/direct/?|
+    tf1.fr/stream/(?P<stream>[\w-]+))
+""", re.VERBOSE))
 class TF1(Plugin):
     api_url = "https://mediainfo.tf1.fr/mediainfocombo/{}?context=MYTF1&pver=4001000"
 
-    def api_call(self, channel, useragent=useragents.CHROME):
-        url = self.api_url.format(f"L_{channel.upper()}")
+    def api_call(self, channel_id, useragent=useragents.CHROME):
+        url = self.api_url.format(channel_id)
         req = self.session.http.get(url,
                                     headers={"User-Agent": useragent})
         return self.session.http.json(req)
```

Comment:
The API calls should ideally use a validation schema instead of simply returning the parsed JSON response and later checking for `if 'delivery' not in data or 'url' not in data['delivery']`. I'm saying this, because `data['delivery']['format']` is not checked for validity either and this will cause a `KeyError`/`LookupError` to be raised if it's missing instead of a proper validation error.